src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -32,6 +32,8 @@ public function __construct(array $decoders = array())
 
     /**
      * {@inheritdoc}
+     * 
+     * @throws RuntimeException If no decoder is found.
```

Comment:
>I was talking about catching subclasses of SerializerException

Ah no I wasn't talking of this one. I find it useful as a doc and helps if you are extending the Serializer in a very coupled way which would break encapsulation. But as I said I personally document it because it's something easily done with PhpStorm at my level and I like saying which exceptions I'm manually throwing, but I perfectly understand that it's not something everyone want to do and would be unmaintainable at a Symfony scale project.

>@throws SerializerException means only that serializer failed.

Precisely, whereas if you only has `\Exception` you have no idea of *what* failed. Well, to put it another why, there reason why not documenting it bothers me is that I don't assume each function call I'm making throws a `\Exception` exception (unless it's a critical part of the app where no failure is allowed). And if an exception is thrown, unless documented, I would expect it to be because something really unusual/weird that need to be fixed immediately. So the current behaviour make it very *intolerant*: Serialization failure is a critical and unexpected error. Thing is, serialization failure is quite common, and unless you dive in the source code, you might get caught by surprised to get an exception at this step, hence this piece of doc to warn the user that this can fail, that it's something you should account for.